standalone-metastore/metastore-common/src/test/java/org/apache/hadoop/hive/common/metrics/metrics2/TestCodahaleMetrics.java

```diff
@@ -74,7 +74,9 @@ public static void setUp() throws Exception {
     conf.set(MetastoreConf.ConfVars.METRICS_CLASS.getHiveName(), CodahaleMetrics.class.getCanonicalName());
     conf.set(MetastoreConf.ConfVars.HIVE_CODAHALE_METRICS_REPORTER_CLASSES.getHiveName(),
         "org.apache.hadoop.hive.common.metrics.metrics2.JsonFileMetricsReporter, "
-            + "org.apache.hadoop.hive.common.metrics.metrics2.JmxMetricsReporter");
+            + "org.apache.hadoop.hive.common.metrics.metrics2.JmxMetricsReporter, "
+                + "org.apache.hadoop.hive.common.metrics.metrics2.Metrics2Reporter, "
```

Comment:
Metrics2Reporter was added twice, is in intentional?

Reply:
Yes, adding the Metrics2Reporter twice causes the exception that is mentioned in the Jira ticket. And this is handled as part of this PR

Reply:
Adding Metrics2Reporter twice would test this scenario.

Reply:
please create a separate unit test for this, it's quite confusing that we intentionally add configs to a unit test setup in a bad way to confirm everything is working fine: a unit test should be explicit about what to do and why

around the new unit test (and the code fix), it should be clarified that:
a) in what circumstances the reporter can be added twice (+who adds that?)
b) how is the patch fixing that?
c) why fixing it here (by a shutdown) is better than making sure that it's not added twice somewhere else

Reply:
Okay I will add a separate unit test for this.